packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue

```diff
@@ -0,0 +1,393 @@
+                        :href="copy.linkHref"
+                        :class="$style['c-cookieBanner-link']"
+                        target="_blank">
+                        {{ copy.linkText }}</a>{{ copy.text4 }}
+                </p>
+            </div>
+
+            <div :class="$style['c-cookieBanner-CTA']">
+                <button-component
+                    tabindex="0"
+                    role="button"
+                    is-full-width
+                    @click.native="acceptActions">
+                    {{ copy.acceptButtonText }}
+                </button-component>
+                <button-component
+                    tabindex="0"
+                    role="button"
+                    button-type="ghost"
+                    is-full-width
+                    @click.native="nonAcceptActions">
+                    {{ copy.nonAcceptButtonText }}
+                </button-component>
+            </div>
+        </div>
+        <div v-if="legacyBanner">
+            <div :class="[$style['c-cookieWarning'], { [$style['c-cookieBanner--hidden']]: hideBanner }]">
+                <div :class="$style['c-cookieWarning-inner']">
+                    <p>
+                        {{ copy.legacyBannerText }}
+                        <a
+                            class="c-cookieWarning-link"
+                            :href="copy.linkHref">
+                            {{ copy.legacyBannerLinkText }}
+                        </a>
+                    </p>
+                    <button
+                        class="c-cookieWarning-btn"
+                        data-test-id="cookieBanner-close-button"
+                        aria-label="Close"
+                        @click="hideBanner = true" />
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import { globalisationServices } from '@justeat/f-services';
+
+import ButtonComponent from '@justeat/f-button';
+import '@justeat/f-button/dist/f-button.css';
+import tenantConfigs from '../tenants';
+
+export default {
+    name: 'CookieBanner',
+
+    components: {
+        ButtonComponent
+    },
+
+    props: {
+        locale: {
+            type: String,
+            default: 'en-GB'
+        },
+
+        isHidden: {
+            type: Boolean,
+            default: false
+        },
+
+        showLegacyBanner: {
+            type: Boolean,
+            default: false
+        },
+
+        cookieExpiry: {
+            type: Number,
+            default: 7776000
+        }
+    },
+
+    data () {
+        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
+        const localeConfig = tenantConfigs[locale];
+        const theme = globalisationServices.getTheme(locale);
+
+        return {
+            copy: { ...localeConfig },
+            theme,
+            hideBanner: false,
+            legacyBanner: false
+        };
+    },
+
+    watch: {
+        isHidden (newVal) {
```

Comment:
Do you need these two watchers? Looks like they're props that determine the visibility of things when they're passed in on render rather than something happening in the component itself?

Reply:
As discussed in Slack...